src/functions/module_financial.ts

```diff
@@ -32,6 +38,340 @@ function newtonMethod(
   return x;
 }
 
+// -----------------------------------------------------------------------------
+// COUPDAYS
+// -----------------------------------------------------------------------------
+export const COUPDAYS: AddFunctionDescription = {
+  description: _lt("Days in coupon period containing settlement date."),
+  args: args(`
+        settlement (date) ${_lt(
+          "The settlement date of the security, the date after issuance when the security is delivered to the buyer."
+        )}
+        maturity (date) ${_lt(
+          "The maturity or end date of the security, when it can be redeemed at face, or par value."
+        )}
+        frequency (number) ${_lt(
+          "The number of interest or coupon payments per year (1, 2, or 4)."
+        )}
+        day_count_convention (number, default=${DEFAULT_DAY_COUNT_CONVENTION} ) ${_lt(
+    "An indicator of what day count method to use."
+  )}
+    `),
+  returns: ["NUMBER"],
+  compute: function (
+    settlement: PrimitiveArgValue,
+    maturity: PrimitiveArgValue,
+    frequency: PrimitiveArgValue,
+    dayCountConvention: PrimitiveArgValue = DEFAULT_DAY_COUNT_CONVENTION
+  ): number {
+    dayCountConvention = dayCountConvention || 0;
+    const start = Math.trunc(toNumber(settlement));
+    const end = Math.trunc(toNumber(maturity));
+    const _frequency = Math.trunc(toNumber(frequency));
+    const _dayCountConvention = Math.trunc(toNumber(dayCountConvention));
+
+    checkMaturityAndSettlementDates(start, end);
+    checkCouponFrequency(_frequency);
+    checkDayCountConvention(_dayCountConvention);
+
+    // https://wiki.documentfoundation.org/Documentation/Calc_Functions/COUPDAYS
+    if (_dayCountConvention === 1) {
+      const before = COUPPCD.compute(settlement, maturity, frequency, dayCountConvention) as number;
+      const after = COUPNCD.compute(settlement, maturity, frequency, dayCountConvention) as number;
+      return after - before;
+    }
+
+    const daysInYear = _dayCountConvention === 3 ? 365 : 360;
+    return daysInYear / _frequency;
+  },
+};
+
+// -----------------------------------------------------------------------------
+// COUPDAYBS
```

Comment:
The alphabetical order between function names can be kept :)
COUPDAYBS > COUPDAYS

Reply:
Switching the to is mughty annoying wiht a rebase with 35 commits. Will not be done, or done in a later commit

Reply:
okay :)